src/plugin/fieldViews/ProseMirrorFieldView.ts

```diff
@@ -346,25 +351,29 @@ export abstract class ProseMirrorFieldView extends FieldView<string> {
   }
 
   protected applyDecorationsFromOuterEditor(
-    decorations: DecorationSource,
-    node: Node,
-    elementOffset: number
+    decorations: DecorationSource | DecorationSet | DecorationGroup,
+    fieldNode: Node
   ) {
     // Do nothing if the decorations have not changed.
     if (decorations === this.outerDecorations) {
       return;
     }
 
     this.outerDecorations = decorations;
-    const localDecoSet = Array.isArray(decorations)
-      ? DecorationSet.create(node, decorations)
-      : decorations;
-    // Offset because the node we are displaying these decorations in is a child of its parent (-1)
-    const localOffset = -1;
-    const offsetMap = new Mapping([
-      StepMap.offset(-elementOffset + localOffset),
-    ]);
-    this.decorations = localDecoSet.map(offsetMap, node);
+    // The incoming decorations are positioned relative to the parent NodeView, so we only
+    // need to consider the offset of this field from its parent NodeView, and can ignore
+    // the offset of the parent from the outer editor's document.
+    // The node we will be altering is a child of its parent, so we add +1 to the offset.
```

Comment:
todo: ask Jon what this line means

Reply:
@jonathonherbert - what does "The node we will be altering is a child of its parent" mean?

Reply:
Maybe better to comment on the original comment, which is also a weird gnomic tautology 😅 : _Offset because the node we are displaying these decorations in is a child of its parent (-1)_

That's not clear. IIUC the whole offset situation looks something like:

> We are creating a DecorationSet local to the node that represents our FieldView, so we must add two offsets:
> - an offset for the position difference between the element received by the NodeView and this field (`elementOffset` – probably better called `elementToFieldOffset`, below `e2foffset`),
> - an offset to align the decorations within the child editor, because the node we are rendering in the child editor is now the `doc` node for that editor.

Why does _the node we are rendering in the child editor is now the `doc` node for that editor_ matter?

If the outer editor sees:

```xml
<doc>
    <!-- other nodes -->
    <element-node>
        <!-- maybe other nodes -->
        <field-node>
            <!-- field content -->
        </field-node>
    </element-node>
</doc>
```

which to ProseMirror's positioning system looks like (abbreviated)

```xml
<doc> <!-- snip --> <element-node> <field-node> <!-- field content, etc. -->
     |             |              |            |
     0             e-offset       e2f-offset   e2f-offset + 1
```

The inner editor sees

```xml
<doc> <!-- previously field-node -->
    <!-- field content -->
</doc>
```

which, if it's passed decorations straight from the parent editor, looks like 

```xml
<doc> <!-- field content, etc. -->
     |             
     e2f-offset + 1         
```

That can't be right, because the position after `doc` within an editor is always 0!

So we must subtract `e2f-offset` to account for the offset from element to field, and subtract -1 to account for the fact that as the `field-content` node becomes a `doc` node, it is removed from the positioning system (there is no 'before' `<doc>`.)

This is I think what I meant by the 'child of its parent' comment above – we're collapsing the child/parent position with this offset – but if the above is clearer, happy to add it to the code in a little PR.

Reply:
(Could also add the renaming in that PR, I think the offset names account for some general confusion here.)

Reply:
Sounds good to me. I prefer `elementToFieldOffset` over `e2foffset`. We don't pay Github per letter.

Reply:
👍  used `e2foffset` in diagrams b/c it fits in a single line – I'll go to the trouble of adding manual line breaks, but only because it's you, @davidfurey.